src/ast/astBuilder.ts

```diff
@@ -305,13 +383,42 @@ export class ASTBuilder implements CVisitor<any> {
 
   visitConditionalExpression(ctx: ConditionalExpressionContext): Expression {
     const logicalOrExpression = ctx.logicalOrExpression();
-    if (logicalOrExpression !== undefined) {
-      return this.visitLogicalOrExpression(logicalOrExpression);
+    if (logicalOrExpression === undefined) {
+      throw new UnreachableCaseError();
     }
+    let expression = this.visitLogicalOrExpression(logicalOrExpression);
```

Comment:
If we are calling the variables below `consequentExpression` and `alternateExpression`, then should we call this variable `predicate` also? Or should we stick to their antlr names when we call `ctx.sth()`? This is because the error message confused me for a moment.

Reply:
Hmm, this could possibly be the predicate for the conditional expression, but it could also just be a standalone expression. I agree that we should stick to the ANTLR names when calling `ctx.sth()` though for consistency.

Are you okay with my changes in https://github.com/CS4215-GN-IY/c-slang/pull/18/commits/7bd6f8d008e8032782c9ab4de46453ef2fe40ead?